extension.js

```diff
@@ -370,7 +403,7 @@ function update_window_geometry() {
 
     if (current_window.maximized_vertically && target_rect.height < workarea.height) {
         Main.wm.skipNextEffect(current_window.get_compositor_private());
-        current_window.unmaximize(Meta.MaximizeFlags.VERTICAL);
+        unmaximize_window(current_window);
```

Comment:
`unmaximize_window()` expects to be called as `notify::maximized-vertically` handler - that's why it accepts the window as an argument and validates it. Please just add `window-maximized` check into the surrounding `if` and call `.unmaximize` directly.

Also, this code currently will call `Main.wm.skipNextEffect()` but won't `.unmaximize()` if `window-maximized` is `true`. So it will incorrectly disable an unrelated effect.